### PR TITLE
Fix hanging `trainer.test()`

### PR DESCRIPTION
**Pull request by amogkam**
Closes https://github.com/ray-project/ray_lightning/issues/132

Fixes an issue where `trainer.test()` hangs when using multiple workers in the test DataLoader.

This issue is a bit weird as I was only able to reproduce the hanging with that exact setup. It does not occur with `trainer.train()` and multiple workers in the training DataLoader. It also does not occur when I set the `num_workers` for the test DataLoader to 0.

I'm not exactly sure what's going on, but the testing actually finishes and the program hangs at `torch.distributed.destroy_process_group()`. It may potentially be related to https://github.com/pytorch/pytorch/issues/75097. The difference between `trainer.train()` and `trainer.test()` is that the former wraps the model in DDP while the latter doesn't (but still creates the process group).

In any case, the `shudown_remote` cleanup code is not actually necessary- the CUDA cache cleanup is already being called in the parent `DDPSpawnPlugin` on each worker, and it seems that `torch.distributed.destroy_process_group()` is not a Public API (and is not being called by PyTorch Lightning either).

The test added in this PR hangs prior to the changes to `ray_ddp.py`, but is passing after.

### CODE REVIEW

File: ray_lightning/tests/test_ddp.py [hunk 1]
```diff
@@ -227,6 +227,15 @@ def test_train_client(tmpdir, start_ray_client_server_2_cpus, num_workers):
     train_test(trainer, model)
 
 
+def test_test_multiple_dataloader_workers(tmpdir, ray_start_2_cpus, seed):
+    """Tests trainer.test with multiple workers for data loading."""
```
Comment (bveeramani) [inline ray_lightning/tests/test_ddp.py:231]: I'm confused by this test. It seems like `BoringModel` uses the `test_dataloader`, which uses 1 worker. But, the test docstring says that multiple workers are used for data loading.

What am I missing?

**Review by bveeramani — COMMENTED** [review]
Had a question. Otherwise looks good to me.

File: ray_lightning/tests/test_ddp.py (hunk 1 above)
Comment (amogkam) [reply ray_lightning/tests/test_ddp.py:231]: Ah sorry it should be >0 workers, not "multiple workers". Will update the comment.

**Review by bveeramani — APPROVED** [review]
LGTM

**Pull request merged**